src/Catalog.API/Extensions/Extensions.cs

```diff
@@ -31,20 +31,18 @@ public static void AddApplicationServices(this IHostApplicationBuilder builder)
 
         if (builder.Configuration["AI:Ollama:Endpoint"] is string ollamaEndpoint && !string.IsNullOrWhiteSpace(ollamaEndpoint))
         {
-            builder.Services.AddEmbeddingGenerator<string, Embedding<float>>(b => b
+            builder.Services.AddEmbeddingGenerator(new OllamaEmbeddingGenerator(ollamaEndpoint, builder.Configuration["AI:Ollama:EmbeddingModel"]))
                 .UseOpenTelemetry()
                 .UseLogging()
-                .Use(new OllamaEmbeddingGenerator(
-                    new Uri(ollamaEndpoint),
-                    builder.Configuration["AI:Ollama:EmbeddingModel"])));
+                .Build();
         }
         else if (!string.IsNullOrWhiteSpace(builder.Configuration.GetConnectionString("openai")))
         {
             builder.AddOpenAIClientFromConfiguration("openai");
-            builder.Services.AddEmbeddingGenerator<string, Embedding<float>>(b => b
+            builder.Services.AddEmbeddingGenerator(s => s.GetRequiredService<OpenAIClient>().AsEmbeddingGenerator(builder.Configuration["AI:OpenAI:EmbeddingModel"]!))
```

Comment:
(super nit)

```suggestion
            builder.Services.AddEmbeddingGenerator(sp => sp.GetRequiredService<OpenAIClient>().AsEmbeddingGenerator(builder.Configuration["AI:OpenAI:EmbeddingModel"]!))
```

Using `sp` is kind of a convention to help people know what this is.